### PR TITLE
Feature/261 agent policy mgmt

### PR DESCRIPTION
**Pull request by weyrick**
Implements #261

### CODE REVIEW

**Review by dscabral — COMMENTED** [review]
Nice, the main idea it's to create a scraper that get states and metrics from backends and send them to the nats via rpc calls...
Also make sense to change the name of functions on policies and fleet... cause we just have some comms when we deal with dataset and not with agents

I leave some comments about the in memory policy repo just to understand what do you have in mind.

**Review by dscabral — APPROVED** [review]
Nice, the main idea it's to create a scraper that get states and metrics from backends and send them to the nats via rpc calls...
Also make sense to change the name of functions on policies and fleet... cause we just have some comms when we deal with dataset and not with agents

I leave some comments about the in memory policy repo just to understand what do you have in mind.

File: agent/policyMgr/manager.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+	repo policies.PolicyRepo
+}
+
+func (a *policyManager) GetPolicyState() ([]policies.PolicyData, error) {
+	d, e := a.repo.GetAll()
+	return d, e
+}
+
+func New(logger *zap.Logger, c config.Config, db *sqlx.DB) (PolicyManager, error) {
```
[the remark this answers is not in the document]
Comment (weyrick) [reply agent/policyMgr/manager.go:35]: yes we may write this out to local sqlite db later to continue policy collection even when cloud connection is interrupted

File: agent/policies/repo.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+package policies
+
+import (
+	"errors"
+	"go.uber.org/zap"
+)
+
+type PolicyRepo interface {
```
[the remark this answers is not in the document]
Comment (weyrick) [reply agent/policies/repo.go:12]: yes, although I'm not entirely sure it's necessary yet, we will see

**Comment by weyrick** [comment]
thanks!

**Pull request merged**